packages/api/src/incentives/transaction-stream/transactionStreamLoop.ts

```diff
@@ -82,26 +89,59 @@ export class TransactionStreamLoopService extends Effect.Service<TransactionStre
               );
               const uniqueTransactions = Array.from(transactionMap.values());
 
+              // Check if there's a SurgeUpgradeEvent in this batch
+              const upgradeTransactionIndex =
+                yield* findFirstSurgeUpgradeTransaction(uniqueTransactions);
+
+              let transactionsToProcess = uniqueTransactions;
+              let shouldStopBatch = false;
+
+              if (upgradeTransactionIndex !== -1) {
+                // Process only transactions up to and including the upgrade transaction
+                transactionsToProcess = uniqueTransactions.slice(
```

Comment:
lets say a batch is 100 txs and the surge tx is at index 0. will the 99 other txs be omitted?